src/v/storage/log_manager.cc

```diff
@@ -55,10 +55,15 @@ using logs_type = absl::flat_hash_map<model::ntp, log_housekeeping_meta>;
 log_manager::log_manager(log_config config, kvstore& kvstore) noexcept
   : _config(std::move(config))
   , _kvstore(kvstore)
-  , _jitter(_config.compaction_interval)
+  , _jitter(_config.compaction_interval())
   , _batch_cache(config.reclaim_opts) {
     _compaction_timer.set_callback([this] { trigger_housekeeping(); });
     _compaction_timer.rearm(_jitter());
+
+    _config.compaction_interval.watch([this]() {
```

Comment:
When I added `watch` (the interface isn't new in this PR), I thought about that -- but it doesn't simplify code much, because anyone watching would still need some RAII handle object to hook into the configuration property and get updates (which would basically be a `binding` without the value).

Reply:
Maybe I don't understand but I see it as the update being propagated into the callback. Like here in this example:
```
_config.compaction_interval.watch([this](int interval){
  _jitter = simple_time_jitter<ss::lowres_clock>(interval);
});
```

Reply:
Ok, I think I misunderstood your "never need to hold a pointer" as proposing a broader structural change.  So yep, it would be slightly terser if the watch interface passed new values in, but I think it's orthogonal to the changes in this PR -- changing that interface & updating all the places that use it would be a separate thing.

Reply:
Gotcha, makes sense